src/Mod/PartDesign/Gui/TaskHelixParameters.h

```diff
@@ -0,0 +1,133 @@
+ *                                                                         *
+ ***************************************************************************/
+
+
+#ifndef GUI_TASKVIEW_TaskHelixParameters_H
+#define GUI_TASKVIEW_TaskHelixParameters_H
+
+#include <Gui/TaskView/TaskView.h>
+#include <Gui/Selection.h>
+#include <Gui/TaskView/TaskDialog.h>
+
+#include "TaskSketchBasedParameters.h"
+#include "ViewProviderHelix.h"
+
+class Ui_TaskHelixParameters;
+
+namespace App {
+class Property;
+}
+
+namespace Gui {
+class ViewProvider;
+}
+
+namespace PartDesignGui {
+
+
+
+class TaskHelixParameters : public TaskSketchBasedParameters
+{
+    Q_OBJECT
+
+public:
+    TaskHelixParameters(ViewProviderHelix *HelixView,QWidget *parent = 0);
+    ~TaskHelixParameters();
+
+    /**
+     * @brief fillAxisCombo fills the combo and selects the item according to
+     * current value of revolution object's axis reference.
+     * @param forceRefill if true, the combo box will be completely refilled. If
+     * false, the current value of revolution object's axis will be added to the
+     * list (if necessary), and selected. If the list is empty, it will be refilled anyway.
+     */
+    void fillAxisCombo(bool forceRefill = false);
+    void addAxisToCombo(App::DocumentObject *linkObj, std::string linkSubname, QString itemText);
+
+private Q_SLOTS:
+    void onPitchChanged(double);
+    void onHeightChanged(double);
+    void onTurnsChanged(double);
+    void onAngleChanged(double);
+    void onAxisChanged(int);
+    void onLeftHandedChanged(bool);
+    void onReversedChanged(bool);
+    void onModeChanged(int);
+
+
+protected:
+    void onSelectionChanged(const Gui::SelectionChanges& msg) override;
+    void changeEvent(QEvent *e) override;
+    bool updateView() const;
+    void getReferenceAxis(App::DocumentObject *&obj, std::vector<std::string> &sub) const;
+
+    double getPitch(void) const;
+    double getHeight(void) const;
+    double getTurns(void) const;
+    double getAngle(void) const;
+    bool getLeftHanded(void) const;
+    bool getReversed(void) const;
+
+    //mirrors of helixes's properties
+    App::PropertyLength*  propPitch;
+    App::PropertyLength*  propHeight;
+    App::PropertyFloat*   propTurns;
+    App::PropertyBool*    propLeftHanded;
+    App::PropertyBool*    propReversed;
+    App::PropertyLinkSub* propReferenceAxis;
+    App::PropertyAngle*   propAngle;
+    App::PropertyInteger* propMode;
+
+
+private:
+    void updateUI();
+
+private:
+    QWidget* proxy;
+    Ui_TaskHelixParameters* ui;
+
+    /**
+     * @brief axesInList is the list of links corresponding to axis combo; must
+     * be kept in sync with the combo. A special value of zero-pointer link is
+     * for "Select axis" item.
+     *
+     * It is a list of pointers, because properties prohibit assignment. Use new
+     * when adding stuff, and delete when removing stuff.
+     */
+    std::vector<App::PropertyLinkSub*> axesInList;
```

Comment:
Again, consider unique_ptr:
```suggestion
    std::vector<std::unique_ptr<App::PropertyLinkSub>> axesInList;
```